mono/metadata/cominterop.c

```diff
@@ -3664,3 +3664,52 @@ ves_icall_System_Runtime_InteropServices_Marshal_FreeBSTR (gpointer ptr)
 {
 	mono_free_bstr (ptr);
 }
+
+void*
+mono_cominterop_get_com_interface (MonoObject *object, MonoClass *ic, MonoError *error)
+{
+#ifndef DISABLE_COM
+	mono_error_init (error);
```

Comment:
I'd use the `error_init(error)` macro rather than the `mono_error_init()` function (I think we marked mono_error_init "external only" so the runtime actually _has_ to use the macro).

Also move the error init outside of the ifdef: if COM is disabled, we still want the error in a valid state (I realize it's unlikely anyone would call this function with COM disabled, but...)